mne/bem.py

```diff
@@ -332,3 +336,109 @@ def cost_fun(x, points):
+    env = os.environ.copy()
+
+    if not os.environ.get('FREESURFER_HOME'):
+        raise RuntimeError('FREESURFER_HOME environment variable not set')
+    if subject:
+        env['SUBJECT'] = subject
+    else:
+        raise RuntimeError('The subject argument must be set')
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    env['SUBJECTS_DIR'] = subjects_dir
+
+    subject_dir = op.join(subjects_dir, subject)
+    mri_dir = op.join(subject_dir, 'mri')
+    T1_dir = op.join(mri_dir, volume)
+    T1_mgz = op.join(mri_dir, volume + '.mgz')
+    bem_dir = op.join(subject_dir, 'bem')
+    ws_dir = op.join(subject_dir, 'bem', 'watershed')
+
+    if not op.exists(subject_dir):
+        raise RuntimeError('Could not find the MRI data directory "%s"'
+                           % subject_dir)
+    if not op.exists(bem_dir):
+        os.makedirs(bem_dir)
+    if not op.exists(T1_dir) and not op.exists(T1_mgz):
+        raise RuntimeError('Could not find the MRI data')
+    if op.exists(ws_dir):
+        if not overwrite:
+            raise RuntimeError('%s already exists. Use the --overwrite option'
+                               'to recreate it.' % ws_dir)
+        else:
+            shutil.rmtree(ws_dir)
+    # put together the command
+    cmd = ['mri_watershed']
+    if preflood:
+        cmd += ["-h",  "%s" % int(preflood)]
+
+    if gcaatlas:
+        cmd += ['-atlas', '-T1', '-brain_atlas', env['FREESURFER_HOME'] +
+                '/average/RB_all_withskull_2007-08-08.gca',
+                subject_dir + '/mri/transforms/talairach_with_skull.lta']
+    elif atlas:
+        cmd += ['-atlas']
+    if op.exists(T1_mgz):
+        cmd += ['-useSRAS', '-surf', op.join(ws_dir, subject), T1_mgz,
+                op.join(ws_dir, 'ws')]
+    else:
+        cmd += ['-useSRAS', '-surf', op.join(ws_dir, subject), T1_dir,
+                op.join(ws_dir, 'ws')]
+    # report and run
+    logger.info('\nRunning mri_watershed for BEM segmentation with the '
+                'following parameters:\n\n'
+                'SUBJECTS_DIR = %s\n'
+                'SUBJECT = %s\n'
+                'Result dir = %s\n' % (subjects_dir, subject, ws_dir))
+    os.makedirs(op.join(ws_dir, 'ws'))
+    run_subprocess(cmd, env=env, stdout=sys.stdout)
+    #
+    os.chdir(ws_dir)
+    if op.exists(T1_mgz):
+        surfaces = [subject + '_brain_surface', subject +
+                    '_inner_skull_surface', subject + '_outer_skull_surface',
+                    subject + '_outer_skin_surface']
+        for s in surfaces:
+            cmd = ['mne_convert_surface', '--surf', s, '--mghmri', T1_mgz,
+                   '--surfout', s]
+            run_subprocess(cmd, env=env, stdout=sys.stdout)
+    os.chdir(bem_dir)
+    if op.exists(subject + '-head.fif'):
+        os.remove(subject + '-head.fif')
+    cmd = ['mne_surf2bem', '--surf', op.join(ws_dir,
+           subject + '_outer_skin_surface'), '--id', '4', '--fif',
+           subject + '-head.fif']
```

Comment:
FYI this is soon to be replaced with a Python version

should I add a Python version of `mne_convert_surface` while I'm at it?